src/binfmt/binfmt.c

```diff
@@ -191,7 +191,8 @@ static int run(int argc, char *argv[]) {
         if (r <= 0)
                 return r < 0 ? EXIT_FAILURE : EXIT_SUCCESS;
 
-        log_setup_service();
+        log_parse_environment();
+        log_open();
```

Comment:
That's fairly ugly... What about going in the other direction: making log_setup_service have a parameter like log_to_console and putting `true` or `false` in the callers?